ehr2meds/PREMEDS/preprocessing/io/dataloader.py

```
@@ -40,32 +40,212 @@ def load_dataframe(
-
-    def _load_csv(
-        self, file_path: str, cols: Optional[List[str]] = None
+        # Try all encodings to read the first line
+        first_line = None
+        for encoding in self.CSV_ENCODINGS:
+            try:
+                with open(file_path, "r", encoding=encoding) as f:
+                    first_line = f.readline()
+                    break  # Successfully read, exit loop
+            except (UnicodeDecodeError, Exception):
+                continue  # Try next encoding
```

Ruff (0.15.0)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)